server/src/main/java/org/opensearch/action/support/replication/ReplicationModeAwareOverrideProxy.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.support.replication;
+
+import org.opensearch.cluster.routing.ShardRouting;
+import org.opensearch.index.seqno.ReplicationTracker.ReplicationMode;
+import org.opensearch.index.shard.ReplicationGroup.ReplicationModeAwareShardRouting;
+
+import java.util.Objects;
+
+/**
+ * This implementation of {@link ReplicationProxy} fans out the replication request to current shard routing basis
+ * the shard routing's replication mode and replication override policy.
+ *
+ * @opensearch.internal
+ */
+public class ReplicationModeAwareOverrideProxy<ReplicaRequest> extends ReplicationProxy<ReplicaRequest> {
+
+    private final ReplicationMode replicationModeOverride;
+
+    public ReplicationModeAwareOverrideProxy(ReplicationMode replicationModeOverride) {
```

Comment:
nit: Lets simply call it `ReplicationModeAwareProxy`

Reply:
made change.